src/main/java/com/diffplug/gradle/InputStreamCollector.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.gradle;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.nio.charset.Charset;
+import java.util.Objects;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.diffplug.common.collect.ImmutableList;
+
+public class InputStreamCollector extends Thread {
+
+	private final InputStream iStream;
+
+	private final PrintStream pStream;
+
+	private final Charset charset;
+
+	private final ImmutableList.Builder<String> output;
+
+	private volatile IOException exception;
+
+	public InputStreamCollector(@Nonnull InputStream is, @Nullable PrintStream ps, @Nullable Charset cs) {
```

Comment:
I think we assume that things are `@Nonnull` by default, similarly to how Guava and Checker Framework approach things, so I'd vote to have this instance of `@Nonnull` removed.

Reply:
I'm OK with that.